plugin/storage/badger/spanstore/writer.go

```diff
@@ -70,7 +77,12 @@ func (w *SpanWriter) WriteSpan(_ context.Context, span *model.Span) error {
 	entriesToStore = append(entriesToStore, trace)
 	entriesToStore = append(entriesToStore, w.createBadgerEntry(createIndexKey(serviceNameIndexKey, []byte(span.Process.ServiceName), startTime, span.TraceID), nil, expireTime))
 	entriesToStore = append(entriesToStore, w.createBadgerEntry(createIndexKey(operationNameIndexKey, []byte(span.Process.ServiceName+span.OperationName), startTime, span.TraceID), nil, expireTime))
-
+	kind, _ := span.GetSpanKind()
+	kindString := strconv.FormatInt(int64(rune(kind)), 10)
+	// This format will convert length of service name to formatted 3-digit number (string) like for 9 it will change to "009"
+	formattedLengthOfService := fmt.Sprintf("%03d", len(span.Process.ServiceName))
```

Comment:
why is this needed?

Reply:
I am trying to get services, operation name and kind from the single index. Kind will be the last character of the key but we have to differentiate between service name and operation name, so storing the length of service name. Currently I am thinking to change the key to `len(serviceName)+"L"+serviceName+operation name+kind`